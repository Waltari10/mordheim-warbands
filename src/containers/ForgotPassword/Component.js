import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import InputAdornment from '@material-ui/core/InputAdornment';
import { makeStyles } from '@material-ui/core/styles';
import { Link as RouterLink, Redirect } from 'react-router-dom';
import EmailIcon from '@material-ui/icons/EmailOutlined';
import InputBase from '@material-ui/core/InputBase';
import * as constants from '../../constants';

import AppWindow from '../../components/AppWindow';

const useStyles = makeStyles((theme) => ({
  header: {
    marginTop: theme.spacing(0),
    marginBottom: theme.spacing(12),
  },
  email: {
    background: 'white',
    borderTopRightRadius: '4px',
    borderTopLeftRadius: '4px',
    height: '48px',
    overflow: 'hidden',
    borderWidth: '0.5px',
    borderStyle: 'solid',
    borderColor: '#C0C0C0',
  },
  continueButton: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
  },
  loginLink: {
    marginBottom: theme.spacing(3),
  },
  content: {
    width: '100%',
  },
  errorMsg: {
    marginBottom: theme.spacing(3),
  },
  inputRoot: {
    backgroundColor: 'white',
    height: '48px',
  },
  input: {
    height: '48px',
    padding: '27px 12px 24px',
  },
  inputAdornment: {
    color: theme.palette.primary.main,
  },
  inputStartAdornment: {
    color: theme.palette.primary.main,
    marginLeft: theme.spacing(2),
  },
}));


const Login = ({
  isAuthorized, sendResetPasswordEmailRequestState, sendResetPasswordEmail,
}) => {

  const [email, setEmail] = useState('');

  const isDisabled = (
    sendResetPasswordEmailRequestState === constants.LOADING ||
    sendResetPasswordEmailRequestState === constants.SUCCESS
  );

  const submitDisabled = isDisabled || !email;

  const classes = useStyles();

  return (
    <AppWindow size="xs">

      {
        isAuthorized &&
          <Redirect
            to='/'
          />
      }

      <Grid className={classes.content} alignItems="center" direction="column" container>
        <Grid className={classes.header} item>
          <Typography variant="h4">Mordheim Warbands</Typography>
        </Grid>
        <Grid item>
          <InputBase
            style={{ display: 'flex' }}
            className={classes.email}
            onChange={(e) => setEmail(e.target.value)}
            // label="Type your email"
            disabled={isDisabled}
            placeholder="email"
            value={email}
            variant="filled"
            startAdornment={
              <InputAdornment position="start">
                <EmailIcon className={classes.inputStartAdornment} />
              </InputAdornment>
            }
          />
        </Grid>
        <Grid item>
          <Button
            className={classes.continueButton}
            disabled={submitDisabled}
            onClick={() => sendResetPasswordEmail(email)}
            variant="contained"
            color="primary"
          >Send password reset email</Button>
        </Grid>

        {
          sendResetPasswordEmailRequestState === constants.ERROR && (
            <Grid className={classes.errorMsg} item>
              <Typography color="error" variant="body1">Something went wrong!</Typography>
            </Grid>
          )
        }
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            component={RouterLink}
            to={'/'}
          >
            <Typography variant="body2">
              Go back to login
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </AppWindow>
  );
};
 
export default Login;