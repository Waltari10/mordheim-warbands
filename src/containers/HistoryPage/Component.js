import React from 'react';

// TODO: Add indicator indicating how far user is in reflcetion
// Add 5xwhy


const HistoryPage = () => {
  return (<div>HistoryPage</div>);
};
 
export default HistoryPage;