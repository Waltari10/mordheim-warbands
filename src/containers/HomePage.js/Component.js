import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { Link as RouterLink } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Typography } from '@material-ui/core';

import PrivateRoute from '../PrivateRoute';
import ReflectionModal from '../ReflectionModal';

import ReflectionListItem from './ReflectionListItem';
import AppWindow from '../../components/AppWindow';


const useStyles = makeStyles((theme) => ({
  listItem: {
    
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    overflow: 'hidden',
  },
  container: {
    marginTop: '53px',
    width: '100%',
    height: '100%',
    placeSelf: 'flex-start',
    overflowY: 'scroll',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    position: 'absolute',
    right: theme.spacing(3),
    bottom: theme.spacing(3),
  },
  addIcon: {
    color: 'white',
    marginTop: '6px',
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '53px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: '#9e9e9e',
    top: 0,
  },
  title: {
    lineHeight: '53px',
  },
  menuIcon: {

    position: 'absolute',
    top: '5px',
    right: theme.spacing(3),
  },
}));


const HomePage = ({
  isLoading, reflections = {}, getReflections, match, logout, uid,
  
}) => {

  useEffect(() => {
    if (uid) {
      getReflections();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <AppWindow className={classes.window} size="xs">

      <div
        className={classes.topNavigationContainer}
      >
        <Typography className={classes.title} align="center" variant="h6">
            Reflection log
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
        >
          <MenuIcon />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
        >
          <MenuItem onClick={() => {
            handleClose();
            logout();
          }}>Logout</MenuItem>
        </Menu>

        
      </div>


      <div className={classes.container}>
        <PrivateRoute component={ReflectionModal} path="/reflection/:id" />
        {isLoading && <Progress className={classes.progress} />}
        {!isLoading && Object.keys(reflections).map((key) => {
          const reflection = reflections[key];

          return (
            <ReflectionListItem
              match={match}
              key={key}
              id={key}
              reflection={reflection}
            />
          );
        })}

        {!isLoading && Object.keys(reflections).length === 0 && (
          <Typography>No reflections yet. Add first reflection.</Typography>
        )}

        
      </div>


      <Fab
        className={classes.hero}
        color="primary"
      >
        <RouterLink
          to={`${match.path}reflection/new`}
        >
          <AddIcon className={classes.addIcon}/>
        </RouterLink>
      </Fab>
    </AppWindow>
  );
};
 
export default HomePage;