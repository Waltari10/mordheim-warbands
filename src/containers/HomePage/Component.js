import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { Redirect } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Typography } from '@material-ui/core';


import * as constants from '../../constants';
import PrivateRoute from '../PrivateRoute';
import WarbandModal from '../WarbandModal';

import WarbandListItem from './WarbandListItem';


const useStyles = makeStyles((theme) => ({
  fabProgress: {
    marginTop: '7px',
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
    backgroundColor: theme.palette.background,
  },
  container: {
    margin: '0 auto',
    display: 'block',
    marginTop: '48px',
    width: '100%',
    maxWidth: '1024px',
    height: 'calc(100% - 48px)',
    placeSelf: 'flex-start',
    overflowY: 'scroll',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    display: 'block',
    margin: '0 auto',
    marginTop: theme.spacing(6),
    marginBottom: theme.spacing(6),
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '48px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: theme.palette.primary.main,
    top: 0,
  },
  title: {
    lineHeight: '48px',
    color: 'white',
  },
  menuIcon: {
    position: 'absolute',
    top: '0px',
    right: theme.spacing(2),
  },
  addIcon: {
    color: 'white',
    marginTop: '7px',
  },
  menu: {
    backgroundColor: 'white',
  },
}));


const HomePage = ({
  isLoading, warbands = {}, getWarbands, match, logout, uid,
  warbandsIndex = [], addWarband, lastAddedWarbandId,
  addWarbandRequestState,
}) => {

  useEffect(() => {
    if (uid) {
      getWarbands();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div className={classes.window}>

      {
        (addWarbandRequestState === constants.SUCCESS && lastAddedWarbandId) && (
          <Redirect
            to={`${match.path}warband/${lastAddedWarbandId}`}
          />)
      }

      <div
        className={classes.topNavigationContainer}
      >
        <Typography className={classes.title} align="center" variant="h5">
            Mordheim Warbands
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
        >
          <MenuIcon
            className={classes.whiteMenuIcon}
          />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
          classes={{
            paper: classes.menu,
          }}
        >
          <MenuItem
            onClick={() => {
              handleClose();
              logout();
            }}>Logout</MenuItem>
        </Menu>


      </div>


      <div className={classes.container}>
        <PrivateRoute component={WarbandModal} path="/warband/:id" />
        {<Progress className={classes.progress} />}
        {warbandsIndex.map(key => {
          const warband = warbands[key];

          return (
            <WarbandListItem
              match={match}
              key={key}
              id={key}
              warband={warband}
            />
          );
        })}


        {!isLoading && (
          <Fab
            className={classes.hero}
            disabled={addWarbandRequestState === constants.LOADING}
            color="primary"
            onClick={() => {
              addWarband({});
            }}
          >
            {
              addWarbandRequestState === constants.LOADING ?
                <Progress className={classes.fabProgress}/> :
                <AddIcon className={classes.addIcon} />
            }
          </Fab>
        )}

        {!isLoading && Object.keys(warbands).length === 0 && (
          <Typography>No Warbands yet. Add first warband.</Typography>
        )}
      </div>
    </div>
  );
};

export default HomePage;