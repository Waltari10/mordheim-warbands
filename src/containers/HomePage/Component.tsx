import React, { useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { match as Match, Redirect } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Theme, Typography } from '@material-ui/core';

import CreateWarbandDialog from './components/CreateWarbandDialog';
import * as constants from '../../constants';
import PrivateRoute from '../PrivateRoute';
import WarbandModal from '../WarbandModal/index';
import SupportDialog from '../../components/SupportContact/index';

import WarbandListItem from './WarbandListItem';
import { DispatchProps, StateProps } from './Container';


const useStyles = makeStyles((theme: Theme) => ({
  fabProgress: {
    marginTop: '7px',
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
    backgroundColor: theme.palette.background.default,
  },
  container: {
    margin: '0 auto',
    display: 'block',
    marginTop: '48px',
    width: '100%',
    maxWidth: '1024px',
    height: 'calc(100% - 48px)',
    placeSelf: 'flex-start',
    overflowY: 'auto',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    display: 'block',
    margin: '0 auto',
    marginTop: theme.spacing(6),
    marginBottom: theme.spacing(6),
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '48px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: theme.palette.primary.main,
    top: 0,
  },
  title: {
    lineHeight: '48px',
    color: 'white',
    marginLeft: '48px',
    marginRight: '48px',
    width: 'calc(100% - 96px)',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
  },
  menuIcon: {
    position: 'absolute',
    top: '0px',
    right: theme.spacing(2),
  },
  whiteMenuIcon: {
    color: 'white',
  },
  addIcon: {
    color: 'white',
    marginTop: '7px',
  },
  menu: {
    backgroundColor: 'white',
  },
  addFirstWarband: {
    marginTop: theme.spacing(12),
  },
}));

interface Props {
  match: Match;
}


const HomePage: React.FunctionComponent<StateProps & DispatchProps & Props> = ({
  isLoading, warbands = {}, getWarbands, match, logout, uid,
  warbandsIndex = [], addWarband, lastAddedWarbandId,
  addWarbandRequestState,
}) => {

  useEffect(() => {
    if (uid) {
      getWarbands();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };


  const [isSupportOpen, setIsSupportOpen] = useState(false);
  const [isCreateWarbandOpen, setIsCreateWarbandOpen] = useState(false);

  return (
    <div className={classes.window}>

      <CreateWarbandDialog
        addWarband={addWarband}
        open={isCreateWarbandOpen}
        close={() => setIsCreateWarbandOpen(false)}
      />

      {
        (addWarbandRequestState === constants.SUCCESS && lastAddedWarbandId) && (
          <Redirect
            to={`${match.path}warband/${lastAddedWarbandId}`}
          />)
      }

      <div
        className={classes.topNavigationContainer}
      >
        <Typography id="homepage_header" className={classes.title} align="center" variant="h5">
          Warbands
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
          id="homepage_menu_button"
        >
          <MenuIcon
            className={classes.whiteMenuIcon}
          />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
          classes={{
            paper: classes.menu,
          }}
        >
          <MenuItem
            id="homepage_logout_button"
            onClick={() => {
              handleClose();
              logout();
            }}>Logout</MenuItem>
          <MenuItem
            onClick={() => {
              setIsSupportOpen(true);
            }}>Support/contact</MenuItem>
          {/* <MenuItem
            onClick={() => {
              setIsDisclaimerOpen(true);
            }}>Disclaimer</MenuItem> */}
        </Menu>


      </div>

      <SupportDialog
        open={isSupportOpen}
        handleClose={() => {
          setIsSupportOpen(false);
        }}
      />

      <div className={classes.container}>
        <PrivateRoute component={WarbandModal} path="/warband/:id" />
        {isLoading && <Progress className={classes.progress} />}
        {warbandsIndex.map(key => {
          const warband = warbands[key];

          return (
            <WarbandListItem
              match={match}
              key={key}
              id={key}
              warband={warband}
            />
          );
        })}

        {!isLoading && Object.keys(warbands).length === 0 && (
          <Typography
            variant="h6"
            className={classes.addFirstWarband}
            align="center"
          >No Warbands yet. Add first warband.</Typography>
        )}


        {!isLoading && (
          <Fab
            id="add_warband_button"
            className={classes.hero}
            disabled={addWarbandRequestState === constants.LOADING}
            color="primary"
            onClick={() => {
              setIsCreateWarbandOpen(true);
            }}
          >
            {
              addWarbandRequestState === constants.LOADING ?
                <Progress className={classes.fabProgress} /> :
                <AddIcon className={classes.addIcon} />
            }
          </Fab>
        )}

      </div>
    </div>
  );
};

export default HomePage;