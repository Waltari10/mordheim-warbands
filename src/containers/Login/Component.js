import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';
import Visibility from '@material-ui/icons/VisibilityOutlined';
import VisibilityOff from '@material-ui/icons/VisibilityOffOutlined';
import { makeStyles } from '@material-ui/core/styles';
import { Link as RouterLink, Redirect } from 'react-router-dom';
import EmailIcon from '@material-ui/icons/EmailOutlined';
import PasswordIcon from '@material-ui/icons/VpnKeyOutlined';
import InputBase from '@material-ui/core/InputBase';


import AppWindow from '../../components/AppWindow';

const useStyles = makeStyles((theme) => ({
  header: {
    marginTop: theme.spacing(0),
    marginBottom: theme.spacing(12),
  },
  email: {
    background: 'white',
    borderTopRightRadius: '4px',
    borderTopLeftRadius: '4px',
    height: '48px',
    overflow: 'hidden',
    borderWidth: '0.5px',
    borderBottomWidth: '0.5px',
    borderBottomStyle: 'dashed',
    borderStyle: 'solid',
    borderColor: '#C0C0C0',


  },
  password: {
    marginBottom: theme.spacing(3),
    background: 'white',
    borderBottomRightRadius: '4px',
    borderBottomLeftRadius: '4px',
    borderWidth: '0.5px',
    borderTopWidth: '0px',
    borderStyle: 'solid',
    height: '48px',
    overflow: 'hidden',
    borderColor: '#C0C0C0',
  },
  continueButton: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
  },
  loginLink: {
    marginBottom: theme.spacing(3),
  },
  content: {
    width: '100%',
  },
  errorMsg: {
    marginBottom: theme.spacing(3),
  },
  inputRoot: {
    backgroundColor: 'white',
    height: '48px',
  },
  input: {
    height: '48px',
    padding: '27px 12px 24px',
  },
  inputAdornment: {
    color: theme.palette.primary.main,
  },
  inputStartAdornment: {
    color: theme.palette.primary.main,
    marginLeft: theme.spacing(2),
  },
}));


const Login = ({
  isLoading, isError, isSuccess, loginWithEmail,
  googleSignIn, isAuthorized,
}) => {

  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');
  const [showPassword, setShowPassword] = useState(true);

  const isDisabled = isLoading || isSuccess;

  const submitDisabled = isDisabled || !password || password.length < 7 || !email;

  const classes = useStyles();

  return (
    <AppWindow size="xs">

      {
        isAuthorized &&
          <Redirect
            to='/'
          />
      }

      <Grid className={classes.content} alignItems="center" direction="column" container>
        <Grid className={classes.header} item>
          <Typography variant="h4">Mordheim Warbands</Typography>
        </Grid>
        <Grid item>
          <form acceptCharset="UTF-8">
            <InputBase
              name="email"
              style={{ display: 'flex' }}
              className={classes.email}
              onChange={(e) => setEmail(e.target.value)}
              // label="Type your email"
              disabled={isDisabled}
              placeholder="email"
              type="email"
              value={email}
              variant="filled"
              startAdornment={
                <InputAdornment position="start">
                  <EmailIcon className={classes.inputStartAdornment} />
                </InputAdornment>
              }
            />
            <InputBase
              name="password"
              variant="filled"
              className={classes.password}
              onChange={(e) => setPassword(e.target.value)}
              disabled={isDisabled}
              type={showPassword ? 'text' : 'password'}
              value={password}
              // label="Type your password"
              placeholder="password"
              startAdornment={
                <InputAdornment position="start">
                  <PasswordIcon className={classes.inputStartAdornment}/>
                </InputAdornment>
              }
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={() => setShowPassword(!showPassword)}
                    onMouseDown={() => setShowPassword(!showPassword)}
                  >
                    {showPassword ?
                      (<Visibility className={classes.inputAdornment} />) :
                      (<VisibilityOff className={classes.inputAdornment} />)
                    }
                  </IconButton>
                </InputAdornment>
              }
            />
          </form>
        </Grid>
        <Grid item>
          <Button
            className={classes.continueButton}
            disabled={submitDisabled}
            onClick={() => loginWithEmail(email, password)}
            variant="contained"
            color="primary"
          >Sign in</Button>
        </Grid>

        {
          isError && (
            <Grid className={classes.errorMsg} item>
              <Typography color="error" variant="body1">Something went wrong!</Typography>
            </Grid>
          )
        }

        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            onClick={googleSignIn}
          >
            <Typography variant="body2">
                Continue with Google
            </Typography>
          </Link>
        </Grid>
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            to={'/signup'}
            component={RouterLink}
          >
            <Typography variant="body2">
              New to Mordheim Warbands? Create an account.
            </Typography>
          </Link>
        </Grid>
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            to={'/forgot-password'}
            component={RouterLink}
          >
            <Typography variant="body2">
             Forgot your password?
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </AppWindow>
  );
};

export default Login;