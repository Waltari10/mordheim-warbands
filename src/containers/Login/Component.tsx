import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';
import Visibility from '@material-ui/icons/VisibilityOutlined';
import VisibilityOff from '@material-ui/icons/VisibilityOffOutlined';
import { makeStyles } from '@material-ui/core/styles';
import { Link as RouterLink, Redirect } from 'react-router-dom';
import EmailIcon from '@material-ui/icons/EmailOutlined';
import PasswordIcon from '@material-ui/icons/VpnKeyOutlined';
import InputBase from '@material-ui/core/InputBase';


import AppWindow from '../../components/AppWindow/index';
import * as constants from '../../constants';
import { StateProps, DispatchProps } from './Container';

const useStyles = makeStyles((theme) => ({
  header: {
    marginTop: theme.spacing(0),
    marginBottom: theme.spacing(12),
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
  email: {
    background: 'white',
    borderTopRightRadius: '4px',
    borderTopLeftRadius: '4px',
    height: '48px',
    overflow: 'hidden',
    borderWidth: '0.5px',
    borderBottomWidth: '0.5px',
    borderBottomStyle: 'dashed',
    borderStyle: 'solid',
    borderColor: '#C0C0C0',
  },
  password: {
    marginBottom: theme.spacing(3),
    background: 'white',
    borderBottomRightRadius: '4px',
    borderBottomLeftRadius: '4px',
    borderWidth: '0.5px',
    borderTopWidth: '0px',
    borderStyle: 'solid',
    height: '48px',
    overflow: 'hidden',
    borderColor: '#C0C0C0',
  },
  continueButton: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
  },
  loginLink: {
    paddingLeft: '48px',
    paddingRight: '48px',
    marginBottom: theme.spacing(3),
    cursor: 'pointer',
  },
  content: {
    width: '100%',
  },
  errorMsg: {
    marginBottom: theme.spacing(3),
    paddingLeft: '24px',
    paddingRight: '24px',
  },
  inputRoot: {
    backgroundColor: 'white',
    height: '48px',
  },
  input: {
    height: '48px',
    padding: '27px 12px 24px',
  },
  inputAdornment: {
    color: theme.palette.primary.main,
  },
  inputStartAdornment: {
    color: theme.palette.primary.main,
    marginLeft: theme.spacing(2),
  },
}));


const Login = ({
  isLoading, isError, isSuccess, loginWithEmail,
  googleSignIn, isAuthorized, error, resetUser,
}: StateProps & DispatchProps) => {

  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');
  const [showPassword, setShowPassword] = useState(false);

  React.useEffect(() => resetUser, []);

  const isDisabled = isLoading || isSuccess;

  const submitDisabled = isDisabled || !password || password.length < 7 || !email;

  const classes = useStyles();

  return (
    <AppWindow size="xs">

      {
        isAuthorized &&
        <Redirect
          to='/'
        />
      }

      <Grid className={classes.content} alignItems="center" direction="column" container>
        <Grid className={classes.header} item>
          <Typography align="center" variant="h4">{constants.APP_NAME}</Typography>
        </Grid>
        <Grid item>
          <form acceptCharset="UTF-8">
            <InputBase
              id="email"
              autoComplete="username"
              name="email"
              style={{ display: 'flex' }}
              className={classes.email}
              onChange={(e) => setEmail(e.target.value)}
              disabled={isDisabled}
              placeholder="email"
              value={email}
              // variant="filled"
              startAdornment={
                <InputAdornment position="start">
                  <EmailIcon className={classes.inputStartAdornment} />
                </InputAdornment>
              }
            />
            <InputBase
              id="password"
              autoComplete="current-password"
              name="password"
              // variant="filled"
              className={classes.password}
              onChange={(e) => setPassword(e.target.value)}
              disabled={isDisabled}
              type={showPassword ? 'text' : 'password'}
              value={password}
              placeholder="password"
              startAdornment={
                <InputAdornment position="start">
                  <PasswordIcon className={classes.inputStartAdornment} />
                </InputAdornment>
              }
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={() => setShowPassword(!showPassword)}
                    onMouseDown={() => setShowPassword(!showPassword)}
                  >
                    {showPassword ?
                      (<Visibility className={classes.inputAdornment} />) :
                      (<VisibilityOff className={classes.inputAdornment} />)
                    }
                  </IconButton>
                </InputAdornment>
              }
            />
          </form>
        </Grid>
        <Grid item>
          <Button
            id="signin_button"
            className={classes.continueButton}
            disabled={submitDisabled}
            onClick={() => loginWithEmail(email, password)}
            variant="contained"
            color="primary"
          >Sign in</Button>
        </Grid>

        {
          isError && (
            <Grid className={classes.errorMsg} item>
              <Typography color="error" variant="body1">
                {(error && error.message) || 'Something went wrong!'}
              </Typography>
            </Grid>
          )
        }

        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            onClick={googleSignIn}
          >
            <Typography align="center" variant="body2">
              Continue with Google
            </Typography>
          </Link>
        </Grid>
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            to={'/signup'}
            component={RouterLink}
          >
            <Typography
              align="center"
              variant="body2">
              New to {constants.APP_NAME}? Create an account.
            </Typography>
          </Link>
        </Grid>
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            to={'/forgot-password'}
            component={RouterLink}
          >
            <Typography variant="body2">
              Forgot your password?
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </AppWindow>
  );
};

export default Login;