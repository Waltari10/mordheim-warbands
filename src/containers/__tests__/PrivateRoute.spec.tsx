
import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';

import Component from '../PrivateRoute/component';


describe('PrivateRoute', () => {
  it('Should render (smoke test)', () => {
    const { asFragment } = render(
      // Requires to be wrapped in router because NotFoundPage uses react-router-doms Link element
      // Requires to be wrapped MuiThemeProvider because NotFoundPage is using Theme from material-ui
      // <Provider store={store}>
      <Router>
        <Component />
      </Router>,
      // </Provider>,
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});