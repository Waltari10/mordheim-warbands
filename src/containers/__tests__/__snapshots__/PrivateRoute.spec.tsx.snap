// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Should render (smoke test) 1`] = `<DocumentFragment />`;
